firmware/mcal/stm32f767/periph/adc.h

```diff
@@ -5,36 +5,36 @@
 #define MCAL_STM32F767_PERIPH_ADC_H_
 
 #include <cstdint>
-#include "stm32f7xx_hal.h"
 
-#include "shared/util/peripheral.h"
+#include "shared/periph/adc.h"
+#include "stm32f7xx_hal.h"
 
 namespace mcal::periph {
 
-class ADCInput : public shared::util::Peripheral {
-
+class ADCInput : public shared::periph::ADCInput {
 private:
-	ADC_HandleTypeDef* hadc_;
+    ADC_HandleTypeDef* hadc_;
 
 public:
-	ADCInput(ADC_HandleTypeDef* hadc) : hadc_(hadc) {};
-	
-	void Start() {
-		HAL_ADC_Start(hadc_);
-	}
-	
-	uint32_t Read() {
-		/// @todo should there be a standard output range?
-		/// if so, the conversion would require knowing the resolution and
-		/// alignment settings
-
-		/// @todo change this to be non-blocking, currently just for demo
-		Start();
-		while(HAL_ADC_PollForConversion(hadc_, 1000) != HAL_OK);
-		return HAL_ADC_GetValue(hadc_);
-	}
+    ADCInput(ADC_HandleTypeDef* hadc) : hadc_(hadc){};
+
+    void Start() override {
+        HAL_ADC_Start(hadc_);
+    }
+
+    uint32_t Read() override {
+        /// @todo should there be a standard output range?
+        /// if so, the conversion would require knowing the resolution and
+        /// alignment settings
+
+        /// @todo change this to be non-blocking, currently just for demo
+        Start();
+        while (HAL_ADC_PollForConversion(hadc_, 1000) != HAL_OK)
+            ;
```

Comment:
Why is the `;` on a new line here? Guessing it's a clang-format thing?

Reply:
Yes it's a clang issue that doesn't have a supported solution. That line of code is also incorrect so I will fix it altogether.

The issue is you should not repeatedly poll the adc, as this renders the 1000ms timeout useless